aptos-move/block-executor/src/scheduler.rs

```diff
@@ -464,17 +526,18 @@ impl Scheduler {
         let mut status = self.txn_status[txn_idx].lock();
         if let TransactionStatus::Suspended(incarnation, dep_condvar) = &*status {
             *status = TransactionStatus::ReadyToExecute(*incarnation, Some(dep_condvar.clone()));
-        } else {
-            unreachable!();
         }
+        // else {
+        //     unreachable!();
+        // }
     }
 
     /// Set status of the transaction to Executed(incarnation).
     fn set_executed_status(&self, txn_idx: TxnIndex, incarnation: Incarnation) {
         let mut status = self.txn_status[txn_idx].lock();
 
-        // Only makes sense when the current status is 'Executing'.
-        debug_assert!(*status == TransactionStatus::Executing(incarnation));
+        // // Only makes sense when the current status is 'Executing'.
```

Comment:
This is fine regardless of whether we add "Halted" status or not, because when the execution is halted, we should not finish execution (makes sense - Halted is final) and set the status to Executed (that could trigger validations, aborts, etc, no need to even potentially allow that to happen).

But instead of debug_assert, I'd check the status and error!("Setting non-Executing status to Executing") if it's not executing. I think this is the recommended way now.